File: src/main/src/org/wdt/intellijmanager/ManagerMain.java
package org.wdt.intellijmanager;

import org.apache.commons.cli.*;
import org.wdt.intellijmanager.command.CopyVmoptionsCommand;
import org.wdt.intellijmanager.command.PrintJetBrainsToolsCommand;
import org.wdt.intellijmanager.command.PrintVersionCommand;
import org.wdt.intellijmanager.command.SaveConfigFileCommand;
import org.wdt.intellijmanager.utils.ToolboxUtils;
import org.wdt.utils.io.FileUtils;

import java.io.IOException;

public class ManagerMain {
    private static final Options options = new Options();

    public static void main(String[] args) throws ParseException, IOException {
        if (FileUtils.isFileNotExists(ToolboxUtils.getToolboxInstallPath())) {
            throw new IOException("Toolbox is not installed!");
        }
        CommandLineParser commandLineParser = new DefaultParser();
        SaveConfigFileCommand configFile = new SaveConfigFileCommand(options);
        CopyVmoptionsCommand copyVmoptionsCommand = new CopyVmoptionsCommand(options);
        PrintJetBrainsToolsCommand printList = new PrintJetBrainsToolsCommand(options);
        PrintVersionCommand info = new PrintVersionCommand(options);
        CommandLine commandLine = commandLineParser.parse(options, args);
        if (commandLine.hasOption(CopyVmoptionsCommand.getToolsPathCommandOption())) {
            copyVmoptionsCommand.copyVmoptionsFile(commandLine);
        } else if (commandLine.hasOption(SaveConfigFileCommand.getSaveFileOption())) {
            configFile.copyFile(commandLine);
        }
        if (commandLine.hasOption(PrintJetBrainsToolsCommand.getOption())) {
            printList.printJetBrainsTools();
        }
        if (commandLine.hasOption(PrintVersionCommand.getOption())) {
            info.printVersionNumber();
        }
    }

}

File: src/main/src/org/wdt/intellijmanager/ManagerMain.kt
package org.wdt.intellijmanager

import org.apache.commons.cli.CommandLineParser
import org.apache.commons.cli.DefaultParser
import org.apache.commons.cli.Options
import org.apache.commons.cli.ParseException
import org.wdt.intellijmanager.command.*
import org.wdt.intellijmanager.command.AddAllOptions.addAllOptions
import org.wdt.intellijmanager.command.ConfigCommand.Companion.configFile
import org.wdt.intellijmanager.command.ConfigCommand.Companion.configOption
import org.wdt.intellijmanager.command.CopyVmoptionsCommand.Companion.toolsPathCommandOption
import org.wdt.intellijmanager.command.PrintVersionCommand.Companion.getOption
import org.wdt.intellijmanager.command.SaveConfigFileCommand.Companion.saveFileOption
import org.wdt.intellijmanager.objects.ConfigObject
import org.wdt.intellijmanager.utils.ToolboxUtils.getToolboxInstallPath
import org.wdt.utils.gson.Json.getBuilder
import org.wdt.utils.io.isFileNotExists
import org.wdt.utils.io.writeStringToFile
import java.io.IOException

object ManagerMain {
    private val options = Options()

    @Throws(ParseException::class, IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        if (getToolboxInstallPath().isFileNotExists()) {
            throw IOException("Toolbox is not installed!")
        }
        if (configFile.isFileNotExists()) {
            configFile.writeStringToFile(getBuilder().serializeNulls().create().toJson(ConfigObject()))
        }
        val commandLineParser: CommandLineParser = DefaultParser()
        addAllOptions(options)
        val configCommand = ConfigCommand()
        val configFile = SaveConfigFileCommand()
        val copyVmoptionsCommand = CopyVmoptionsCommand()
        val printList = PrintJetBrainsToolsCommand()
        val info = PrintVersionCommand()
        val commandLine = commandLineParser.parse(options, args)
        if (commandLine.hasOption(toolsPathCommandOption)) {
            copyVmoptionsCommand.copyVmoptionsFile(commandLine)
        } else if (commandLine.hasOption(saveFileOption)) {
            configFile.copyFile(commandLine)
        }
        if (commandLine.hasOption(PrintJetBrainsToolsCommand.option)) {
            printList.printJetBrainsTools()
        }
        if (commandLine.hasOption(getOption())) {
            info.printVersionNumber()
        }
        if (commandLine.hasOption(configOption)) {
            configCommand.startTask(commandLine)
        }
    }
}
